src/Microsoft.DotNet.VersionTools/Dependencies/ProjectJsonUpdater.cs

```diff
@@ -29,33 +29,20 @@ public IEnumerable<BuildInfo> Update(IEnumerable<BuildInfo> buildInfos)
         {
             foreach (string projectJsonFile in ProjectJsonPaths)
             {
-                JObject projectRoot;
+                IEnumerable<BuildInfo> buildInfosUsed = null;
+
                 try
                 {
-                    projectRoot = ReadProject(projectJsonFile);
+                    FileUtils.ReplaceFileContents(projectJsonFile, contents => ReplaceAllDependencyVersions(contents, projectJsonFile, buildInfos, out buildInfosUsed));
                 }
                 catch (Exception e)
                 {
-                    Trace.TraceWarning($"Non-fatal exception occurred reading '{projectJsonFile}'. Skipping file. Exception: {e}. ");
+                    Trace.TraceWarning($"Non-fatal exception occurred reading or writing '{projectJsonFile}'. Skipping file. Exception: {e}. ");
```

Comment:
I agree that the JSON-parsing code could throw an exception, but I guess I was thinking that parsing could be considered part of reading, but I can also see using a more narrow definition of file-reading which excludes parsing.  Regardless, I've changed it to "processing".